### ISSUE TITLE
Conflict with #include wpilib.h?

### ISSUE THREAD

**Issue by ghost** [Team2168/2168_Vision_Example#4]
Hello and awesome vision example.
I'm able to compile ftp and run your code on the roborio fine.
Now I'm looking at taking the example Arcade Drive and adding in the include/lib references and am having a issue.
First, I am able to build and run the Arcade Drive example, 
I've added to it:
#include <opencv2/opencv.hpp>
using namespace cv;
I added _Includes\OpenCV\2.4.10 to the includes under paths and symbols in c++ general
I also added _Libraries\softFP\lib_OpenCV and lib_cpp to the library paths and
I also added the opencv_highgui,opencv_core,opencv_imgproc to the libraries list

When I compile, it appears there is a conflict between wpilib.h and opencv.hpp I get tons of redeclaration errors... Here is a sample: 
Note there are lots of these, I've only added 4...Below them is the full code, Its not long.
Thanks for any help!
Eric

08:06:48 **** Incremental Build of configuration Debug for project Arcade Drive ****
Info: Internal Builder is used for build
arm-frc-linux-gnueabi-g++ -std=c++1y "-IC:\\Users\\user/wpilib/cpp/current/include" "-IC:\\Users\\user\\Desktop\\FRC2017Stuff\\workspace\\Arcade Drive\\src" "-IC:\\Users\\user/wpilib/user/cpp/include" "-IC:\\Users\\user\\Desktop\\FRC2017Stuff\\2168_Vision_Example\\_Includes\\OpenCV\\2.4.10" -O0 -Og -g3 -Wall -c -fmessage-length=0 -o "src\\Robot.o" "..\\src\\Robot.cpp" 
In file included from C:\Users\user\Desktop\FRC2017Stuff\2168_Vision_Example\_Includes\OpenCV\2.4.10/opencv2/opencv.hpp:59:0,
                 from ..\src\Robot.cpp:4:
C:\Users\user\Desktop\FRC2017Stuff\2168_Vision_Example\_Includes\OpenCV\2.4.10/opencv2/contrib/contrib.hpp:852:22: error: redefinition of 'class cv::LDA'
     class CV_EXPORTS LDA
                      ^
In file included from C:\Users\user/wpilib/cpp/current/include/opencv2/core/types_c.h:101:0,
                 from C:\Users\user/wpilib/cpp/current/include/opencv2/core/core_c.h:48,
                 from C:\Users\user\Desktop\FRC2017Stuff\2168_Vision_Example\_Includes\OpenCV\2.4.10/opencv2/opencv.hpp:46,
                 from ..\src\Robot.cpp:4:
C:\Users\user/wpilib/cpp/current/include/opencv2/core.hpp:2422:18: error: previous definition of 'class cv::LDA'
 class CV_EXPORTS LDA
                  ^
In file included from C:\Users\user\Desktop\FRC2017Stuff\2168_Vision_Example\_Includes\OpenCV\2.4.10/opencv2/opencv.hpp:59:0,
                 from ..\src\Robot.cpp:4:
C:\Users\user\Desktop\FRC2017Stuff\2168_Vision_Example\_Includes\OpenCV\2.4.10/opencv2/contrib/contrib.hpp:968:27: error: redeclaration of 'COLORMAP_AUTUMN'
         COLORMAP_AUTUMN = 0,
                           ^
In file included from C:\Users\user/wpilib/cpp/current/include/opencv2/imgproc/imgproc.hpp:48:0,
                 from C:\Users\user\Desktop\FRC2017Stuff\2168_Vision_Example\_Includes\OpenCV\2.4.10/opencv2/opencv.hpp:50,
                 from ..\src\Robot.cpp:4:
C:\Users\user/wpilib/cpp/current/include/opencv2/imgproc.hpp:3782:5: note: previous declaration 'cv::ColormapTypes COLORMAP_AUTUMN'
     COLORMAP_AUTUMN = 0, //!< ![autumn](pics/colormaps/colorscale_autumn.jpg)
     ^
In file included from C:\Users\user\Desktop\FRC2017Stuff\2168_Vision_Example\_Includes\OpenCV\2.4.10/opencv2/opencv.hpp:59:0,
                 from ..\src\Robot.cpp:4:

FULL Program :
using namespace std;
#include <stdio.h>
#include <stdlib.h>
#include <opencv2/opencv.hpp>
#include <string>
#include <ctime>
#include <iostream>
#include <iomanip>
#include <sstream>
#include <unistd.h>
#include <pthread.h>

using namespace cv;

#include <WPILib.h>

class Robot: public frc::SampleRobot {
	frc::RobotDrive myRobot { 0, 1 };  // robot drive system
	frc::Joystick stick { 0 };  // only joystick
public:
	Robot() {
		myRobot.SetExpiration(0.1);
	}

	/**
	 * Runs the motors with arcade steering.
	 */
	void OperatorControl()
	{
		while (IsOperatorControl() && IsEnabled()) {
		    double x= stick.GetRawAxis(0);
			myRobot.ArcadeDrive(stick); // drive with arcade style (use right stick)
			frc::Wait(0.005);			// wait for a motor update time
		}
	}

};

START_ROBOT_CLASS(Robot)

**Comment by ghost** [comment]
It appears some includes in my paste didnt make it, heres the ones between opencv.hpp and unistd.h
#include <string>
#include <ctime>
#include <iostream>
#include <iomanip>
#include <sstream>

**Comment by ghost** [comment]
Note I put WPILib.h down at the bottom in an attempt to prevent the errors, it was the first thing in the code, also, the using namespace std; as the first line, if its moved down, results in errors finding vector in opencv.hpp

**Comment by NotInControl** [comment]
Let me investigate, I believe new in 2017 wpilib uses openCV for its camera stuff, and so both wpilib.h and opencvcore.hpp are importing the same named libraries.

If this is true, try modifying the imports so only one is being imported



Happy Connecting. Sent from my Sprint Samsung Galaxy S® 5


-------- Original message --------

**Comment by ghost** [comment]
Yes, when I comment out the <opencv2/opencv.hpp> it compiles partly, I get a conflict with the constructor for vcap.open (I think you made a custom version)
and it does not know what CV_CAP_PROP_EXPOSURE_ABSOLUTE is...

should I just look at using opencv from wpilib and change the constructor?  Howimportant is the absolute exposure setting?

**Comment by NotInControl** [comment]
yes you can use the wpi lib and avoid the exposure. It is not important if you do not care about it. It was something we did to support the Microsoft Lifecam for other teams which was provided by FRC back in 2015. 

To get this up and running on your end, you don't need it
